### PR TITLE
optionals to pointers + tests

### LINKED ISSUE #1170: In the API types, optional structs should be pointers.
Some API fields are optional, but they should then be pointers as per:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required

## Expected Behavior

There's couple of structs in the APIs that are marked as optional, but they are not pointers. This means that they get filled in as empties and that causes validation errors.
https://github.com/kedacore/keda/blob/v2/api/v1alpha1/triggerauthentication_types.go#L25
https://github.com/kedacore/keda/blob/v2/api/v1alpha1/triggerauthentication_types.go#L34

And that I could then define (simplified example):
```
		Spec: kedav1alpha1.TriggerAuthenticationSpec{
			SecretTargetRef: []kedav1alpha1.AuthSecretTargetRef{
				{
					Parameter: "host",
					Name:      secretName,
					Key:       secretKey,
				},
			},
```

## Actual Behavior
However, unless I add this here, it will not pass because the webhook will fail it.
```
		Spec: kedav1alpha1.TriggerAuthenticationSpec{
			SecretTargetRef: []kedav1alpha1.AuthSecretTargetRef{
				{
					Parameter: "host",
					Name:      secretName,
					Key:       secretKey,
				},
			},
			// HACK. Without this doesn't work.
			HashiCorpVault: kedav1alpha1.HashiCorpVault{Secrets: []kedav1alpha1.VaultSecret{}},
```

## Steps to Reproduce the Problem

  1. Call the API with that object, webhook will fail it saying that VaultSecret is null
  2. But, now I'm then left with these kinds of log messages:
```
2020-09-17T18:50:07.737Z	ERROR	scalehandler	Error authenticate to Vault	{"type": "ScaledObject", "namespace": "default", "name": "ping-trigger", "triggerAuthRef.Name": "default-trigger-auth", "error": "Vault auth method  is not supported"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.1/zapr.go:128
github.com/kedacore/keda/pkg/scaling/resolver.ResolveAuthRef
	/workspace/pkg/scaling/resolver/scale_resolvers.go:74
github.com/kedacore/keda/pkg/scaling.(*scaleHandler).buildScalers
	/workspace/pkg/scaling/scale_handler.go:306
github.com/kedacore/keda/pkg/scaling.(*scaleHandler).GetScalers
	/workspace/pkg/scaling/scale_handler.go:67
github.com/kedacore/keda/pkg/scaling.(*scaleHandler).checkScalers
	/workspace/pkg/scaling/scale_handler.go:182
github.com/kedacore/keda/pkg/scaling.(*scaleHandler).startScaleLoop
	/workspace/pkg/scaling/scale_handler.go:134
```

But at least I can now create the resource :) 
  3.

## Specifications

  - **KEDA Version:** V2 Head
  - **Platform & Version:** *Please elaborate*
  - **Kubernetes Version:** *Please elaborate*
  - **Scaler(s):** *Please elaborate*

### PR DESCRIPTION
**Pull request by vaikas**
Signed-off-by: Ville Aikas <vaikas@vmware.com>

_Provide a description of what has been changed_

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)
- [X] Tests have been added
- [X] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs
- [X] Changelog has been updated

Fixes #1170 

There are no user facing documents, since from the users point of view they are the same, just the clients changed. Also, add omitempty as it was missing from some of these. Add tests.

Since I'm not very familiar with the code style yet, I was curious about maybe changing the ResolveAuthRef to actually return an error in addition to map[string]string, string so it would be: (map[string]string, string, error) but didn't want to make that change without discussion first. I think that would be better for the callers to have an error to deal with rather than missing information?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,7 @@
 
 ### Improvements
 
+- Change API optional structs to pointers to conform with k8s guide ([#1170](https://github.com/kedacore/keda/issues/1170))
```
Comment (zroubalik) [inline CHANGELOG.md:38]: Could you please move this to `###Other` section below, as this is not something that is directly affecting users.

**Review by zroubalik — APPROVED** [review]
LGTM, minor nit in Changelog

**Comment by zroubalik** [comment]
@ahmelsayed could you please take a look on the question? I think that it does make sense.

File: CHANGELOG.md (hunk 1 above)
Comment (vaikas) [reply CHANGELOG.md]: Sure thing, done :)

**Comment by ahmelsayed** [comment]
@vaikas I think I didn't want `ResolveAuthRef` to cause a reconciliation error. if it could resolve the auth params, great, otherwise the scaler will fail later on if it indeed needed those params. In some services we try to reduce hard error paths and limit errors to the parts that need it. I'm okay with returning an error though as I think the user here might find it more useful to catch those configuration errors early rather than having to hunt through each scaler's error logs.

**Review by ahmelsayed — APPROVED** [review]

**Comment by ahmelsayed** [comment]
If it's okay with you, we can merge this change and have that other change as a separate PR

**Comment by vaikas** [comment]
Yea, I'm fine with merging as is and doing more refactoring later on if we think it's warranted :) Just wanted to throw it out there. I can take a spin to see what it might look like and then bikeshed on that one :)

**Pull request merged**